app/ledger/neonLedger.js

```diff
@@ -252,9 +265,10 @@ export const getPublicKeys = async (
 export const getDeviceInfo = async () => {
   const ledger = await NeonLedger.init()
   try {
-    const deviceInfo = await ledger.getDeviceInfo()
-    const publicKey = await ledger.getPublicKey()
-    return { deviceInfo, publicKey }
+    return await Promise.all([
+      ledger.getDeviceInfo(),
+      ledger.getPublicKey(),
+    ]).then(([deviceInfo, publicKey]) => ({ deviceInfo, publicKey }))
```

Comment:
See comment above about cleaning this up

Reply:
resolved in b49c341178a97650cf73ab078f3d2b95fb16e521